tests/InfillTest.cpp

```diff
@@ -0,0 +1,221 @@
+            pattern(pattern),
+            zig_zagify(zig_zagify),
+            connect_polygons(connect_polygons),
+            line_distance(line_distance)
+        {
+            name = makeName("InfillParameters_%d_%d_%d_%lld", (int)pattern, (int)zig_zagify, (int)connect_polygons, line_distance);
+        }
+    };
+
+    class InfillTestParameters
+    {
+    public:
+        bool valid;  // <-- if the file isn't read (or anything else goes wrong with the setup) we can communicate it to the tests
+        std::string fail_reason;
+        size_t test_polygon_id;
+
+        // Parameters used to generate the infill:
+        InfillParameters params;
+        Polygons outline_polygons;
+
+        // Resulting infill:
+        Polygons result_lines;
+        Polygons result_polygons;
+
+        std::string name;
+
+        InfillTestParameters() :
+            valid(false),
+            fail_reason("Read of file with test polygons failed (see generateInfillTests), can't continue tests."),
+            test_polygon_id(-1),
+            params(InfillParameters(EFillMethod::NONE, false, false, 0)),
+            outline_polygons(Polygons()),
+            result_lines(Polygons()),
+            result_polygons(Polygons()),
+            name("UNNAMED")
+        {
+        }
+
+        InfillTestParameters(const InfillParameters& params, const size_t& test_polygon_id, const Polygons& outline_polygons, const Polygons& result_lines, const Polygons& result_polygons) :
+            valid(true),
+            fail_reason("__"),
+            test_polygon_id(test_polygon_id),
+            params(params),
+            outline_polygons(outline_polygons),
+            result_lines(result_lines),
+            result_polygons(result_polygons)
+        {
+            name = makeName("InfillTestParameters_P%d_Z%d_C%d_L%lld__%lld", (int)params.pattern, (int)params.zig_zagify, (int)params.connect_polygons, params.line_distance, test_polygon_id);
+        }
+
+        friend std::ostream& operator<<(std::ostream& os, const InfillTestParameters& params)
+        {
+            return os << params.name << "(" << (params.valid ? std::string("input OK") : params.fail_reason) << ")";
+        }
+    };
+
+    constexpr coord_t outline_offset = 0;
+    constexpr coord_t infill_line_width = 350;
+    constexpr coord_t infill_overlap = 0;
+    constexpr size_t infill_multiplier = 1;
+    const AngleDegrees fill_angle = 0.;
+    constexpr coord_t z = 100; // Future improvement: Also take an uneven layer, so we get the alternate.
+    constexpr coord_t shift = 0;
+	const std::vector<std::string> polygon_filenames =
+		{
+			"../tests/resources/polygon_concave.txt",
+			"../tests/resources/polygon_concave_hole.txt",
+			"../tests/resources/polygon_square.txt",
+			"../tests/resources/polygon_square_hole.txt",
+			"../tests/resources/polygon_triangle.txt",
+			"../tests/resources/polygon_two_squares.txt",
+		};
```

Comment:
Using tabs here instead of spaces. I think at some point your IDE switched.

Reply:
Yeah, it's because of the new laptop. Fixed!